src/planetary_system.jl

```diff
@@ -352,6 +352,340 @@ function generate_period_and_sizes_power_law(s::Star, sim_param::SimParam; num_p
+    @assert length(P) == length(mass) == length(ecc)
+    @assert star_mass > 0
+    @assert all(P .> 0)
+    @assert all(mass .> 0)
+    @assert all(1 .> ecc .>= 0)
+
+    if length(P) > 0
+        P_segments_unstable = compute_unstable_Hill_regions_periods_given_planets(P, mass, insert_pl_mass::Float64, star_mass, sim_param; ecc=ecc, verbose=verbose)
+        P_draw = draw_lognormal_allowed_regions(P_segments_unstable; μ=μ, σ=σ, x_min=x_min, x_max=x_max)
+    else
+        P_draw = invert_cdf_lognormal(rand(Uniform(cdf_lognormal(x_min; μ=μ, σ=σ), cdf_lognormal(x_max; μ=μ, σ=σ))); μ=μ, σ=σ)
+    end
+    return P_draw
+end
+
+function draw_period_lognormal_allowed_regions_mutualHill(P::AbstractVector{Float64}, mass::AbstractVector{Float64}, insert_pl_mass::Float64, star_mass::Float64, sim_param::SimParam; μ::Float64=0., σ::Float64=1., x_min::Float64=0., x_max::Float64=Inf, ecc::AbstractVector{Float64}=zeros(length(P)), insert_pl_ecc::Float64=0., verbose::Bool=false)
+    @assert length(P) == length(mass) == length(ecc)
+    @assert 0 < insert_pl_mass < star_mass
+    @assert all(P .> 0)
+    @assert all(mass .> 0)
+    @assert all(1 .> ecc .>= 0)
+    @assert 0 <= insert_pl_ecc < 1
+
+    if length(P) > 0
+        P_segments_unstable = compute_unstable_mutualHill_regions_periods_given_planets(P, mass, insert_pl_mass, star_mass, sim_param; ecc=ecc, insert_pl_ecc=insert_pl_ecc, verbose=verbose)
+        P_draw = draw_lognormal_allowed_regions(P_segments_unstable; μ=μ, σ=σ, x_min=x_min, x_max=x_max)
+    else
+        P_draw = invert_cdf_lognormal(rand(Uniform(cdf_lognormal(x_min; μ=μ, σ=σ), cdf_lognormal(x_max; μ=μ, σ=σ))); μ=μ, σ=σ)
+    end
+    return P_draw
+end
+
+function draw_periodscale_power_law_allowed_regions_mutualHill(pl_per_cl::Vector{Int64}, P_cl::AbstractVector{Float64}, mass_cl::AbstractVector{Float64}, insert_cl_ρ::AbstractVector{Float64}, insert_cl_mass::AbstractVector{Float64}, star_mass::Float64, sim_param::SimParam, x0::Float64, x1::Float64, α::Float64; ecc_cl::AbstractVector{Float64}=zeros(length(P_cl)), insert_cl_ecc::AbstractVector{Float64}=zeros(length(insert_cl_mass)), verbose::Bool=false)
+    @assert x0 < x1
+    @assert sum(pl_per_cl) == length(P_cl) == length(mass_cl) == length(ecc_cl)
+    @assert all(P_cl .> 0)
+    @assert all(mass_cl .> 0)
+    @assert all(0 .<= ecc_cl .< 1)
+    @assert length(insert_cl_ρ) == length(insert_cl_mass) == length(insert_cl_ecc)
+    @assert all(0 .< insert_cl_ρ)
+    @assert all(0 .< insert_cl_mass .< star_mass)
+    @assert all(0 .<= insert_cl_ecc .< 1)
+
+    if length(P_cl) > 0
+        Pc_segments_unstable = compute_unstable_mutualHill_regions_periodscales_given_clusters(pl_per_cl, P_cl, mass_cl, insert_cl_ρ, insert_cl_mass, star_mass, sim_param; ecc_cl=ecc_cl, insert_cl_ecc=insert_cl_ecc, verbose=verbose)
+        Pc_draw = draw_power_law_allowed_regions(Pc_segments_unstable, x0, x1, α)
```

Comment:
Should it do anything to deal with cases of returning a NaN?

Reply:
Similar to setting NaNs for the unscaled periods of planets that cannot be fit into a cluster anymore, I want this function to return a NaN for the period scale for cases where the cluster we are trying to insert cannot fit (so the planets in this cluster have their periods all set to NaNs, as we have been originally doing).

Reply:
Ok. I'd suggest adding that as a comment, so we'll remember that in the future.

Reply:
I've made sure to make this clear in the function documentation strings!

Reply:
Thanks.